Backend/src/main/java/com/example/Backend/controller/UserController.java

```diff
@@ -0,0 +1,136 @@
+
+    @Operation(summary = "Login user", description = "Authenticate user with provided credentials")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "202", description = "User Logged in Successfully!"),
+            @ApiResponse(responseCode = "404", description = "User Doesn't Exist!"),
+            @ApiResponse(responseCode = "401", description = "Wrong Password!")
+
+    })
+    @PostMapping("/login")
+    public ResponseEntity<String> loginUser(@Valid @RequestBody UserLogin userLogin, BindingResult bindingResult) {
+        StringBuilder errorMessage = new StringBuilder();
+        try {
+            String loginServiceResponse = this.loginService.login(userLogin);
+            if (loginServiceResponse.equals("Found")) {
+                log.info("User Login: {}", userLogin.getUsername());
+                return new ResponseEntity<>("User Logged in Successfully!", HttpStatus.ACCEPTED);
+            }
+            else {
+                if (bindingResult.hasErrors()) {
+                    List<FieldError> errors = bindingResult.getFieldErrors();
+                    errorMessage = new StringBuilder("Validation errors: ");
+                    for (FieldError error : errors) {
+                        errorMessage.append(error.getDefaultMessage()).append("\n");
+                    }
+                    throw new MethodArgumentNotValidException(null, bindingResult);
+                }
+                else if(loginServiceResponse.equals("Wrong Password"))
+                {
+                    log.info("Wrong Password for user: {}", userLogin.getUsername());
+                    return new ResponseEntity<>("Wrong Password!", HttpStatus.UNAUTHORIZED);
+                }
+                else if(loginServiceResponse.equals("Not Found"))
+                {
+                    log.info("User Doesn't Exists!");
+                    return new ResponseEntity<>("User Doesn't Exist!", HttpStatus.NOT_FOUND);
+                }
+            }
+        } catch (MethodArgumentNotValidException e) {
+            log.error("Validation Errors: {}", errorMessage);
+        }
+        return new ResponseEntity<>(errorMessage.toString(), HttpStatus.NOT_FOUND);
+    }
+
+    @Operation(summary = "Register user", description = "Register a new user")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "User registered successfully!"),
+            @ApiResponse(responseCode = "409", description = "User Exists Already!"),
+            @ApiResponse(responseCode = "400", description = "Insufficient Parameters!")
+    })
+    @PostMapping("/register")
+    public ResponseEntity<String> registerUser(@Valid @RequestBody UserRegister userRegister, BindingResult bindingResult) {
+        StringBuilder errorMessage = new StringBuilder();
+        try {
+            String registerServiceResponse = this.registerService.register(userRegister);
+            if (registerServiceResponse.equals("Registered")) {
```

Comment:
We are breaking the initial principle called Single Responsibility Principle. Do check how we can refactor the code here as It becomes difficult to read with so many if else statements.